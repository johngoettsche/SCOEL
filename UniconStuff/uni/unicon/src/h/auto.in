/* define's for zlib, jpeg, opengl, etc. */

/* define as 1 if we have vfork */
#undef HAVE_WORKING_VFORK

/* define as 1 if we have FTGL */
#undef HAVE_LIBFTGL

/* define as 1 if we have Freetype */
#undef HAVE_LIBFREETYPE

/* define as 1 if we have Xft */
#undef HAVE_LIBXFT

/* define as 1 if we have ogg */
#undef HAVE_LIBOGG

/* define as 1 if we have vorbis */
#undef HAVE_LIBVORBIS

/* define as 1 if we have libsmpeg */
#undef HAVE_LIBSMPEG

/* define as 1 if we have libSDL */
#undef HAVE_LIBSDL

/* define as 1 if we have libopenal */
#undef HAVE_LIBOPENAL

/* define as 1 if we have alut */
#undef HAVE_LIBALUT

/* define as 1 if we have jvoiplib */
#undef HAVE_LIBJVOIP

/* define as 1 if we have jvoiplib */
#undef HAVE_LIBJRTP

/* define as 1 if we have jvoiplib */
#undef HAVE_LIBJTHREAD

/* define as 1 if we have jvoiplib */
#undef HAVE_LIBVOIP

/* define as 1 if we have libsndfile */
#undef HAVE_LIBSNDFILE

/* define as 1 if we have Zlib/libz/whatever */
#undef HAVE_LIBZ

/* define as 1 if we have jpeg */
#undef HAVE_LIBJPEG

/* define as 1 if we have png; disabled if libz is missing or turned off */
#undef HAVE_LIBPNG

/* define as 1 if we have opengl */
#undef HAVE_LIBGL

/* define as 1 if we have openssl */
#undef HAVE_LIBSSL

/* define as 1 if we have odbc */
#undef HAVE_LIBIODBC
#undef HAVE_LIBODBC

/* define as 1 if we have getrlimit */
#undef HAVE_GETRLIMIT

/* define as 1 if we have setrlimit */
#undef HAVE_SETRLIMIT

/* define as 1 if we have a timezone variable */
#undef HAVE_TIMEZONE

/* define as 1 if struct tm has a tm_zone field */
#undef HAVE_STRUCT_TM_TM_ZONE

/* define as 1 if struct tm has a tm_gmtoff field */
#undef HAVE_STRUCT_TM_TM_GMTOFF

/* define as 1 if tzname exists */
#undef HAVE_TZNAME

/* define as 1 if X11 is found */
#undef HAVE_LIBX11

/* define as 1 if strerror exists */
#undef HAVE_STRERROR

/* define as 1 if strerror_r exists */
#undef HAVE_STRERROR_R

/* define as 1 if vfork exists */
#undef HAVE_VFORK

/* define as 1 if sys_nerr exists */
#undef HAVE_SYS_NERR

/* define as 1 if sys_errlist exists */
#undef HAVE_SYS_ERRLIST

/* Define to empty if the keyword does not work.  */
#undef const

/* Define to `unsigned' if <sys/types.h> doesn't define.  */
#undef size_t

/* Define if you have the ANSI C header files.  */
#undef STDC_HEADERS

/* Define to `signed' if not defined in <sys/types.h> */
#undef ssize_t

/* Define if running on Solaris 2.4+ which requires defining
 * __EXTENSIONS__ to get prototypes for snprintf() and strncasecmp() */
#ifndef __EXTENSIONS__
#undef __EXTENSIONS__
#endif

/* Define if running on Linux or other GNU Libc system which requires
 * defining _BSD_SOURCE to get prototypes for... */
#ifndef _BSD_SOURCE
#undef _BSD_SOURCE
#endif

/* Define if you have the bcopy function.  */
#undef HAVE_BCOPY

/* Define if you have the bzero function.  */
#undef HAVE_BZERO

/* Define if you have the inet_aton function.  */
#undef HAVE_INET_ATON

/* Define if you have the inet_pton function.  */
#undef HAVE_INET_PTON

/* Define if you have the memcpy function.  */
#undef HAVE_MEMCPY

/* Define if you have the memset function.  */
#undef HAVE_MEMSET

/* Define if you have the snprintf function.  */
#undef HAVE_SNPRINTF

/* Define if you have the socket function.  */
#undef HAVE_SOCKET

/* Define if you have the vsprintf function.  */
#undef HAVE_VSPRINTF

/* Define if you have the <arpa/inet.h> header file.  */
#undef HAVE_ARPA_INET_H

/* Define if you have the <errno.h> header file.  */
#undef HAVE_ERRNO_H

/* Define if you have the <net/errno.h> header file.  */
#undef HAVE_NET_ERRNO_H

/* Define if you have the <netdb.h> header file.  */
#undef HAVE_NETDB_H

/* Define if you have the <netinet/in.h> header file.  */
#undef HAVE_NETINET_IN_H

/* Define if you have the <strings.h> header file.  */
#undef HAVE_STRINGS_H

/* Define if you have the <sys/errno.h> header file.  */
#undef HAVE_SYS_ERRNO_H

/* Define if you have the <sys/socket.h> header file.  */
#undef HAVE_SYS_SOCKET_H

/* Define if you have the <sys/types.h> header file.  */
#undef HAVE_SYS_TYPES_H

/* Define if you have the <unistd.h> header file.  */
#undef HAVE_UNISTD_H

/* Define if you have the nsl library (-lnsl).  */
#undef HAVE_LIBNSL

/* Define if you have the socket library (-lsocket).  */
#undef HAVE_LIBSOCKET

/* Define if using alloca.c.  */
#undef C_ALLOCA

/* Define to one of _getb67, GETB67, getb67 for Cray-2 and Cray-YMP systems.
   This function is required for alloca.c support on those systems.  */
#undef CRAY_STACKSEG_END

/* Define if you have alloca.h and it should be used (not Ultrix).  */
#undef HAVE_ALLOCA_H

/* Define if your struct stat has st_blksize.  */
#undef HAVE_ST_BLKSIZE

/* Define if int is 16 bits instead of 32.  */
#undef INT_16_BITS

/* Define if long int is 64 bits.  */
#undef LONG_64_BITS

/* Define to `long' if <sys/types.h> doesn't define.  */
#undef off_t

/* If using the C implementation of alloca, define if you know the
   direction of stack growth for your system; otherwise it will be
   automatically deduced at run-time.
	STACK_DIRECTION > 0 => grows toward higher addresses
	STACK_DIRECTION < 0 => grows toward lower addresses
	STACK_DIRECTION = 0 => direction of growth unknown
 */
#undef STACK_DIRECTION

/* Define if your processor stores words with the most significant
   byte first (like Motorola and SPARC, unlike Intel and VAX).  */
#undef WORDS_BIGENDIAN

/* Define if you have flock.  */
#undef HAVE_FLOCK

/* Define if you have fsync.  */
#undef HAVE_FSYNC

/* Define if you have ftruncate.  */
#undef HAVE_FTRUNCATE

/* Define if you have rename.  */
#undef HAVE_RENAME

/* Define if you have the <fcntl.h> header file.  */
#undef HAVE_FCNTL_H

/* Define if you have the <memory.h> header file.  */
#undef HAVE_MEMORY_H

/* Define if you have the <stdlib.h> header file.  */
#undef HAVE_STDLIB_H

/* Define if you have the <string.h> header file.  */
#undef HAVE_STRING_H

/* Define if you have the <sys/file.h> header file.  */
#undef HAVE_SYS_FILE_H

/* Define if you have the dbm library (-ldbm).  */
#undef HAVE_LIBDBM

/* Define if you have the ndbm library (-lndbm).  */
#undef HAVE_LIBNDBM

/* Define if you have the pthreads library (-lpthread).  */
#undef HAVE_LIBPTHREAD

/* Define if you have PTHREAD_MUTEX_RECURSIVE_NP  */
#undef HAVE_PTHREAD_MUTEX_RECURSIVE_NP

/* Define if you have getaddrinfo.  */
#undef HAVE_GETADDRINFO

/* Define if you have clock_gettime.  */
#undef HAVE_CLOCK_GETTIME

/* Define if you have getrusage.  */
#undef HAVE_GETRUSAGE

/* Define if you have <sys/resource.h>  */
#undef HAVE_SYS_RESOURCE_H

/* Define if you have crypt  */
#undef HAVE_CRYPT

/* Define if you have sysconf */
#undef HAVE_SYSCONF

/* Define if you have profil */
#undef HAVE_PROFIL
