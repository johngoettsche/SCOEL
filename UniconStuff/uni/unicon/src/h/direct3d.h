/*
 * Direct 3D port of the 3D facilities. Not implemented yet.
 */

/*
 * Redefinitions of various OpenGL integer scalar values.
 * Using these names, portable code can function without OpenGL.
 * If OpenGL is present, they are defined as OpenGL-native, see opengl.h
 * On other platforms, they may be a simple integer enumeration or may be
 * redefined to values native to that platform.
 */

#define U3D_POINTS         1
#define U3D_LINES          2
#define U3D_LINE_STRIP     3
#define U3D_LINE_LOOP      4
#define U3D_TRIANGLES      5
#define U3D_TRIANGLE_FAN   6
#define U3D_TRIANGLE_STRIP 7
#define U3D_QUADS          8
#define U3D_QUAD_STRIP     9
#define U3D_POLYGON       10

#define U3D_REPLACE      100
#define U3D_BLEND        101
#define U3D_MODULATE     102

#define GLfloat float
#define GLubyte unsigned char
#define GLuint unsigned int

