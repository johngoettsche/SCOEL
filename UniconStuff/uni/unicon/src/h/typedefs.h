/*
 * Type for an externally findable & setable integer, used by "setsize". CS
 */

#if SCCX_MX
   typedef union {
      char magicText[4];
      int  value;
   } setint_t;
#endif 					/* SCCX_MX */

/*
 * typdefs for the run-time system.
 */

typedef AllocType msize;
typedef int ALIGN;		/* pick most stringent type for alignment */
typedef unsigned int DIGIT;

/*
 * Default sizing and such.
 */

/*
 * Set up typedefs and related definitions depending on whether or not
 * ints and pointers are the same size.
 */

#if IntBits != WordBits
#ifdef LongLongWord
   typedef long long int word;
   typedef unsigned long long int uword;
   #else
   typedef long int word;
   typedef unsigned long int uword;
   #endif
#else					/* IntBits != WordBits */
   typedef int word;
   #ifdef CDC_VXVE
      typedef uword;
   #else				/* CDC_VXVE */
      typedef unsigned int uword;
   #endif				/* CDC_VXVE */
#endif					/* IntBits != WordBits */

#ifndef PointerDef
   typedef void *pointer;
#endif					/* PointerDef */

/*
 * Typedefs to make some things easier.
 */

typedef int (*fptr)();
typedef struct descrip *dptr;

typedef word C_integer;

/*
 * A success continuation is referenced by a pointer to an integer function
 *  that takes no arguments.
 */
typedef int (*continuation) (void);

#ifdef PthreadCoswitch
/*
 * Treat an Icon "cstate" array as an array of context pointers.
 * cstate[0] is used by Icon code that thinks it's setting a stack pointer.
 * We use cstate[1] to point to the actual context struct.
 * (Both of these are initialized to NULL by Icon 9.4.1 or later.)
 */
//typedef struct context **cstate;
#endif					/* PthreadCoswitch */

#if !COMPILER

   /*
    * Typedefs for the interpreter.
    */
   
   /*
    * Icode consists of operators and arguments.  Operators are small integers,
    *  while arguments may be pointers.  To conserve space in icode files on
    *  computers with 16-bit ints, icode is written by the linker as a mixture
    *  of ints and words (longs).  When an icode file is read in and processed
    *  by the interpreter, it looks like a C array of mixed ints and words.
    *  Accessing this "nonstandard" structure is handled by a union of int and
    *  word pointers and incrementing is done by incrementing the appropriate
    *  member of the union (see the interpreter).  This is a rather dubious
    *  method and certainly not portable.  A better way might be to address
    *  icode with a char *, but the incrementing code might be inefficient
    *  (at a place that experiences a lot of execution activity).
    *
    * For the moment, the dubious coding is isolated under control of the
    *  size of integers.
    */
   
   #if IntBits != WordBits
   
      typedef union {
         int *op;
         word *opnd;
         } inst;
      
      #else				/* IntBits != WordBits */
      
      typedef union {
         word *op;
         word *opnd;
         } inst;
      
   #endif				/* IntBits != WordBits */
      
#endif					/* COMPILER */
