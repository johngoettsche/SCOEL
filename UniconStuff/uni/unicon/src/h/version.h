/*
 * version.h -- version identification
 */

#undef DVersion
#undef Version
#undef UVersion
#undef IVersion

/*
 *  Icon version number and date.
 *  These are the only two entries that change any more.
 */
#define VersionNumber "13.0"

#define VersionDate "Feb 1, 2017"

/*
 * Version number to insure format of data base matches version of iconc
 *  and rtt.
 */

#define DVersion "12.1.00"

#if COMPILER

   /*
    * &version
    */
   #define Version  "Unicon Version " VersionNumber " (iconc).  " VersionDate

#else					/* COMPILER */

   /*
    *  &version
    */
   #define Version  "Unicon Version " VersionNumber ".  " VersionDate
   
   /*
    * Version numbers to be sure ucode is compatible with the linker
    * and icode is compatible with the run-time system.
    */
   
   #define UVersion "U12.1.00"
   
       #ifdef FieldTableCompression

	  #if IntBits == 16
	     #define IVersion "I12.U.30FT/16/16"
	  #endif				/* IntBits == 16 */

	  #if IntBits == 32
#if WordBits==64
	     #define IVersion "I12.U.30FT/32/64"
#else
	     #define IVersion "I12.U.30FT/32/32"
#endif
	  #endif				/* IntBits == 32 */

	  #if IntBits == 64
	     #define IVersion "I12.U.30FT/64/64"
	  #endif				/* IntBits == 64 */

       #else				/* FieldTableCompression */

	  #if IntBits == 16
	     #define IVersion "I12.U.30/16/32"
	  #endif				/* IntBits == 16 */

	  #if IntBits == 32
#if WordBits==64
	     #define IVersion "I12.U.30/32/64"
#else
	     #define IVersion "I12.U.30/32/32"
#endif
	  #endif				/* IntBits == 32 */

	  #if IntBits == 64
	     #define IVersion "I12.U.30/64/64"
	  #endif				/* IntBits == 64 */

       #endif				/* FieldTableCompression */
   
#endif					/* COMPILER */

/*
 * Version number for event monitoring.
 */
#define Eversion "12.0.00"
