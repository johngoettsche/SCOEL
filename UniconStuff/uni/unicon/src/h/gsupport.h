/*
 * Group of include files for translators, etc. 
 */

#include "../h/define.h"

#if CSET2V2
   #include <io.h>
#endif					/* CSet/2 ver 2 */

#if !VMS && !UNIX	 /* don't need path.h */
   #include "../h/path.h"
#endif					/* !VMS && !UNIX */

#include "../h/config.h"
#include "../h/sys.h"
#include "../h/typedefs.h"
#include "../h/cstructs.h"
#include "../h/proto.h"
#include "../h/cpuconf.h"

#ifdef HAVE_GETADDRINFO
#undef HAVE_GETADDRINFO
#endif 

#ifdef ConsoleWindow
   #include "../h/rmacros.h"
   #include "../h/rstructs.h"
   #include "../h/graphics.h"
   #include "../h/rexterns.h"
   #include "../h/rproto.h"
#endif					/* ConsoleWindow */

#ifdef mdw_Instrument_Allocations
extern void * _alloc(unsigned int, char *, int);
#define alloc(n) (_alloc((n),__FILE__,__LINE__))
#endif /* mdw_Instrument_Allocations */

/* mdw: squelch redef of "OF" as gcc warning */
#ifdef OF
#undef OF
#endif

/* mdw: squelch redef of "Type" as gcc warning */
#ifdef Type
#undef Type
#endif

