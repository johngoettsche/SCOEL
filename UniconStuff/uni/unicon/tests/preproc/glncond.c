#if 0
1
2
#else
3
4
#endif
#include "glncond.h"
9
10
