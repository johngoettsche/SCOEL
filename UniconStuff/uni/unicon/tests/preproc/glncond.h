#if 1
5
6
#else
7
8
#endif
>__LINE__<
