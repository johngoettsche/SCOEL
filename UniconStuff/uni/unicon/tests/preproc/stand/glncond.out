#line 4 "glncond.c"

3
4

#line 1 "glncond.h"

5
6

#line 7 "glncond.h"

>8<

#line 8 "glncond.c"

9
10
